lua/feline/generator.lua

```diff
@@ -328,17 +347,109 @@ function M.generate_statusline(winid)
+            parsed_sections[i] = {}
+
+            for j, component in ipairs(section) do
+                local ok, result = pcall(parse_component, component, winid)
+
+                if not ok then
+                    api.nvim_err_writeln(string.format(
+                        "Feline: error while processing component number %d on section %d "..
+                        "of type '%s': %s",
+                        j, i, statusline_type, result
+                    ))
+
+                    result = { str = '', len = 0 }
+                end
+
+                parsed_sections[i][j] = result
+                component_indices[#component_indices+1] = {i, j}
+                statusline_length = statusline_length + result.len
+            end
+        end
+
+        local win_width = api.nvim_win_get_width(winid)
+
+        -- If statusline length is larger than the window width, sort the component indices
+        -- in ascending order of priority of the components they refer to
+        -- Then truncate the components one by one using by their short_provider or hiding them
+        -- entirely until the statusline fits within the window
+        if statusline_length > win_width then
+            table.sort(component_indices, function(a, b)
+                -- Access the original component through the sections table using the indices
+                return (sections[a[1]][a[2]].priority or 0) < (sections[b[1]][b[2]].priority or 0)
+            end)
+
+            for _, indices in ipairs(component_indices) do
+                -- Access the original and parsed values using the indices
+                local component = sections[indices[1]][indices[2]]
+                local parsed_component = parsed_sections[indices[1]][indices[2]]
+
+                -- If short_provider exists, use it, else hide the component if the truncate_hide
+                -- component value is set to true
+                if component.short_provider then
+                    -- Get new parsed component value using the short_provider and calculate the
+                    -- length difference between the two values, and if it's greater than 0, use
+                    -- the new value instead of the old one
+                    local parsed_component_new = parse_component(component, winid, true)
+                    local length_difference = parsed_component.len - parsed_component_new.len
+
+                    if length_difference > 0 then
+                        -- Update statusline length and replace old parsed value with new one
+                        statusline_length = statusline_length - length_difference
+                        parsed_sections[indices[1]][indices[2]] = parsed_component_new
+                    end
+                elseif component.truncate_hide then
+                    statusline_length = statusline_length - parsed_component.len
+                    parsed_sections[indices[1]][indices[2]] = {str = '', len = 0}
```

Comment:
```suggestion

```

This isn't needed since the check happens after anyways. Also causes an issue when you have a component with no short provider, high priority and truncate_hide set to true. It will still get hidden even though it has a priority higher than a component with a short provider.

Reply:
That isn't supposed to happen because components are always hidden in order of priority. Also the check that happens afterwards hides all components with `truncate_hide` set to true, even if they have a short provider, which should only happen if all other components are using their `short_provider`

Reply:
Well, for one there is no point of checking the `truncate_hide` value for components without a short_provider because they would have been hidden by the code I suggested to removed anyway. Anyway, you should be able to reproduce the bug with this [config](https://gist.github.com/ram02z/34d6f2fae2c1a35b455ea480ea2da726). Just keep making the active smaller and you will see that the higher priority component is still hidden first.

Reply:
> Well, for one there is no point of checking the `truncate_hide` value for components without a short_provider because they would have been hidden by the code I suggested to be removed anyway. Anyway, you should be able to reproduce the bug with this [config](https://gist.github.com/ram02z/34d6f2fae2c1a35b455ea480ea2da726). Just keep making the active smaller and you will see that the higher priority component is still hidden first.

Ah, I see what you mean now. That behavior was intended actually, but thinking more about it, it makes more sense to remove it